libdnf/dnf-context.cpp

```diff
@@ -1582,8 +1588,35 @@ dnf_context_setup_enrollments(DnfContext *context, GError **error)
                                                     NULL);
     g_autoptr(GKeyFile) repofile = rhsm_utils_yum_repo_from_context (rhsm_ctx);
 
-    if (!g_key_file_save_to_file (repofile, repofname, error))
-        return FALSE;
+    bool sameContent = false;
+    do {
+        int fd;
+        if ((fd = open(repofname, O_RDONLY)) == -1)
+            break;
+        gsize length;
+        g_autofree gchar *data = g_key_file_to_data(repofile, &length, NULL);
+        auto fileLen = lseek(fd, 0, SEEK_END);
+        if (fileLen != static_cast<long>(length)) {
+            close(fd);
+            break;
+        }
+        if (fileLen > 0) {
+            std::unique_ptr<char[]> buf(new char[fileLen]);
+            lseek(fd, 0, SEEK_SET);
+            auto readed = read(fd, buf.get(), fileLen);
```

Comment:
This doesn't handle e.g. `EINTR`.  It'd probably be easiest to just `mmap()` the file (`g_mapped_file_new`), then you can conveniently `memcmp()` without reading it into malloc manually.

Or compute a checksum over both and compare the checksums.

Reply:
@cgwalters
* Using mmap is good idea. May be next time. PR was merged.
* I prefer to compare contents than compute checksums. Checksums use resources for computation. And then I need to compare the checksums. Moreover when checksums are the same I may only say that the contents are probably same.  So I have to do comparison or ignore possibility of hash collision.